openedx/core/djangoapps/content_libraries/api/containers.py

```diff
@@ -192,11 +192,18 @@ def get_container_from_key(container_key: LibraryContainerLocator, isDeleted=Fal
     raise ContentLibraryContainerNotFound
 
 
-def get_container(container_key: LibraryContainerLocator, include_collections=False) -> ContainerMetadata:
+def get_container(
+    container_key: LibraryContainerLocator,
+    *,
+    include_collections=False,
+    container: Container | None = None,
+) -> ContainerMetadata:
     """
     Get a container (a Section, Subsection, or Unit).
     """
-    container = get_container_from_key(container_key)
+    if not container:
+        container = get_container_from_key(container_key)
+    assert container.key == container_key.container_id
```

Comment:
This is fine to leave as-is, but I'm personally not a huge fan of this API style of taking an ID and optionally a pre-loaded object.

I would personally suggest:

```python
def get_container(
    container_or_key: LibraryContainerLocator | Container,
    /, *,
    include_collections=False,
) -> ContainerMetadata:
    """
    Get metadata about a container (a Section, Subsection, or Unit).
    """
    container = container_or_key if isinstance(container_or_key, Container) else _get_container_from_key(container_or_key)
```

Reply:
Reverted this, cf [49557f5](https://github.com/openedx/edx-platform/pull/36528/commits/49557f570e45f640188897773fbbaf48f8bc0d04)